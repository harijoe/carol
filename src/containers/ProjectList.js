import React, { PropTypes, Component } from 'react'
import { connect } from 'react-redux'
import { fromProject, fromStatus } from 'store/selectors'
import { projectList, PROJECT_LIST, setToken } from 'store/actions'

import { ProjectList } from 'components'

class ProjectListContainer extends Component {
  static propTypes = {
    list: PropTypes.array,
    loading: PropTypes.bool,
    request: PropTypes.func.isRequired,
  }

  componentWillMount() {
    this.props.request()
  }

  render() {
    const { list, loading } = this.props

    return <ProjectList {...{ list, loading }} />
  }
}

const mapStateToProps = state => ({
  list: fromProject.getList(state),
  loading: fromStatus.isLoading(state, PROJECT_LIST.prefix),
})

const mapDispatchToProps = dispatch => ({
  request: () => {
    setToken(dispatch).then(() => {
      dispatch(projectList.request())
    })
  },
})

export default connect(mapStateToProps, mapDispatchToProps)(ProjectListContainer)
