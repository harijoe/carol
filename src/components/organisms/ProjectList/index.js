import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { FormattedMessage } from 'react-intl'
import { theme, breakpoint, breakpointMax } from 'utils/style'

import { Section, Heading, ProjectListItem, Loading, Grid, Row } from 'components'
import { Carousel } from 'containers'

const StyledSection = styled(Section)`
  position: relative;
  min-height: 100vh;

  &::before {
    position: absolute;
    left: 0;
    top: 0;
    height: 40%;
    width: 100%;
    background: ${theme('colors.white')};
    content: '';

    ${breakpoint('m')`
      height: 32rem;
    `}
  }
`

const StyledHeading = styled(Heading)`
  position: relative;

  ${breakpoint('l')`
    margin-bottom: ${theme('spaces.xxl')};
  `}
`

const StyledGrid = styled(Grid)`
  ${breakpoint('l')`
    max-width: 100rem;
  `}
`

const StyledRow = styled(Row)`
  ${breakpoint('m')`
    margin-left: calc(${theme('spaces.l')} / -2);
    margin-right: calc(${theme('spaces.l')} / -2);

      &.qs-Carousel-Projects > div {
       width: 100%;

       > div {
        display: flex;
        flex-wrap: wrap;

        > div {
          padding: calc(${theme('spaces.l')} / 2);
          max-width: 50%;
          width: 50%;
        }
      }
    }
  `}

  ${breakpoint('l')`
    &.qs-Carousel-Projects > div > div {
      > div {
        max-width: 33.33%;
        width: 33.33%;
      }
    }
  `}
`

const StyledCarousel = styled(Carousel)`
  ${breakpointMax('m')`
    max-width: 100vw;

    .slick-list {
      padding-bottom: ${theme('spaces.m')};
    }
  `}
`

const ProjectList = ({ list, loading }) => (
  <StyledSection light>
    <StyledGrid narrow>
      <StyledHeading level={1}><FormattedMessage id="project.list.heading" /></StyledHeading>
      <Loading loading={loading && list.length === 0}>
        { list.length === 0 && <FormattedMessage id="project.list.empty" /> }
        <StyledRow className="qs-Carousel-Projects">
          <StyledCarousel
            slidesToShow={1}
            slidesToScroll={1}
            variableWidth
            infinite={false}
            dots
            responsive={[
              {
                breakpoint: 767,
                settings: {
                  swipe: true,
                },
              },
              {
                breakpoint: 10000,
                settings: 'unslick',
              },
            ]}
          >
            {list.map(items => (
              <ProjectListItem
                key={items['@id']}
                {...items}
              />
            ))}
          </StyledCarousel>
        </StyledRow>
      </Loading>
    </StyledGrid>
  </StyledSection>
)

ProjectList.propTypes = {
  list: PropTypes.array.isRequired,
  loading: PropTypes.bool,
}

export default ProjectList
