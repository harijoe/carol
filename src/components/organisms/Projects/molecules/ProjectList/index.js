import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { theme, breakpoint, breakpointMax } from 'utils/style'

import { Row } from 'components'
import { Carousel } from 'containers'
import ProjectListItem from '../ProjectListItem'

const StyledRow = styled(Row)`
  ${breakpoint('m')`
    margin-left: calc(${theme('spaces.l')} / -2);
    margin-right: calc(${theme('spaces.l')} / -2);

      &.qs-Carousel-Projects > div {
       width: 100%;

       > div {
        display: flex;
        flex-wrap: wrap;

        > div {
          padding: calc(${theme('spaces.l')} / 2);
          max-width: 50%;
          width: 50%;
        }
      }
    }
  `}

  ${breakpoint('l')`
    &.qs-Carousel-Projects > div > div {
      > div {
        max-width: 33.33%;
        width: 33.33%;
      }
    }
  `}
`

const StyledCarousel = styled(Carousel)`
  ${breakpointMax('m')`
    max-width: 100vw;

    .slick-list {
      padding-bottom: ${theme('spaces.m')};
    }
  `}
`

const ProjectList = ({ list }) =>
  <StyledRow className="qs-Carousel-Projects">
    <StyledCarousel
      slidesToShow={1}
      slidesToScroll={1}
      variableWidth
      infinite={false}
      dots
      responsive={[
        {
          breakpoint: 767,
          settings: {
            swipe: true,
          },
        },
        {
          breakpoint: 10000,
          settings: 'unslick',
        },
      ]}
    >
      {list.map(items => <ProjectListItem key={items['@id']} {...items} />)}
    </StyledCarousel>
  </StyledRow>

ProjectList.propTypes = {
  list: PropTypes.array.isRequired,
}

export default ProjectList
