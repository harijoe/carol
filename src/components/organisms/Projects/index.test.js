import React from 'react'
import { shallow } from 'enzyme'

import mockIntl from '../../../../test/intlMock'
import ProjectList from './'

const list = [{ id: 1, name: 'project 1' }, { id: 2, name: 'project 2' }, { id: 3, name: 'project 3' }]

it('renders the ProjectList component', () => {
  const wrapper = shallow(mockIntl(<ProjectList list={list} />))

  expect(wrapper).toMatchSnapshot()
})
