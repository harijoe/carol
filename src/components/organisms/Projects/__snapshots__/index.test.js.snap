// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ProjectList component 1`] = `
<Projects
  list={
    Array [
      Object {
        "id": 1,
        "name": "project 1",
      },
      Object {
        "id": 2,
        "name": "project 2",
      },
      Object {
        "id": 3,
        "name": "project 3",
      },
    ]
  }
/>
`;
